kv/txn_coord_sender.go

```diff
@@ -501,7 +501,7 @@ func (tc *TxnCoordSender) cleanupTxn(trace *tracer.Trace, txn proto.Transaction)
 	txnMeta, ok := tc.txns[id]
 	// Only clean up once per transaction.
```

Comment:
This comment and the one on the last line of the method are not quite accurate. We may want to remove the `txnEnd` nil check completely (although it does improve the error message in race conditions: `txnEnd` becomes nil during `cleanupTxn`, but `ok` doesn't become false until the heartbeat goroutine has stopped. It's a panic either way, but this panic is nicer than a double-closed channel).